### PR TITLE
Logs: Change logic creating uid in LogRowModel

### LINKED ISSUE #68112: logs: handle duplicate ids
logs datasources can provide an `id` for every log-row. `logsModel.ts` consumes this value, and uses it for various tasks (log-row-context, react-keys in logs-visualization).
it relies on the IDs being unique, otherwise duplicate react-keys happens, and cause problems. (please note: this causes very serious problems, log-lines vanishing etc. not just some kind of warning in the console).

(one situation where IDs can get duplicated is when running two logs-queries in explore, and they return the same log line).

the `logsModel.ts` should detect this problem when consuming the logs-dataframes, and adjust it's internal representation so somehow the duplication is eliminated (for example, start appending something to the id, or something else).

### LINKED ISSUE #68119: loki: more uniform id-generation
(note: first https://github.com/grafana/grafana/issues/68112 has to be implemented)

the loki datasource plugin has to generate IDs for every log-row, because such a field is not available from the loki-database.

the current version of this code adds the `query.refId` to this `id` ( https://github.com/grafana/grafana/blob/18d4fda6132ec5ec4d7de32e3e425c702a8845b5/pkg/tsdb/loki/frame.go#L177 ) 

this causes problems because engineers usually assume the same log-line always creates the same id (like it happens with other logs-databases), and this causes bugs. so whenever we want to use the `id` to match with another query, we have to keep the `query.refId` the same.

why does the loki datasource do this?

because the consumer of this `id` value (`logsModel.ts`), is unable to handle duplicate IDs, which may happen if we run two loki queries in explore which return the same log-line.

we will fix this weakness in `logsModel.ts` in https://github.com/grafana/grafana/issues/68112, and when that's done, we should eliminate the addition of the refId to the loki-id, to make id-generation more predictable.

### LINKED ISSUE #68428: Explore view with elasticsearch plugin: log panel not correctly updated when limit option is changed
### 
# What went wrong?

**What happened**:

I'm querying an elasticsearch datasource with query type `logs` in the explore view. I enabled the feature `enableElasticsearchBackendQuerying`.
When updating the `limit` option from 500 to 10 for example, the logs displayed in the log panel are not updated correctly.  I'm still seeing hundreds of logs. And worse, when I try to click on some log lines, nothing happens.

This also happens when I'm selecting consecutively different time windows in the log volume panel, after a few updates, the log panel doesn't show the logs I'm expected to see.

The current workaround is to refresh the page. After that, everything works as expected.

**What did you expect to happen**:
 
When I update the limit option from 500 to 10, I'm expecting to see only 10 log lines in the log panel.

### 
# How do we reproduce it?

**Step 1:**

- Start Grafana with the feature `enableElasticsearchBackendQuerying` enabled. I suspect it does even without the feature.

**Step 2:**

- Open the explore view
- Select an elasticsearch datasource
- Select the logs query type
- Run first a query with the default limit (default to 500), your query must return several log lines.
- Run a second query with a limit to 1.
- Look at the log panel and see that you have more than 1 log line.

### 
# What Grafana version are you using?

10.1.0-116348pre-ubuntu

### 
### Is the bug inside a Dashboard Panel?

_No response_

### Grafana Platform?

Docker

### User's OS?

macOS Monterey

### User's Browser?

Google Chrome Version 112.0.5615.137 (Official Build) (arm64)

### Is this a Regression?

None

### Are Datasources involved?

Elasticsearch

### Anything else to add?

_No response_

### PR DESCRIPTION
**Pull request by ivanahuckova**
In this PR we are fixing issue with unresponsive log lines/buggy experience in a case when user does 2 queries and some of the log lines have same `id`. We later use this `id` as `uid` and use it as `key` when rendering log lines. 

To make `uid` truly unique, we append `refId` of query that produced this log result.

Fixes: https://github.com/grafana/grafana/issues/68428
Fixes: https://github.com/grafana/grafana/issues/68112
Fixes: https://github.com/grafana/grafana/issues/68119

# Release notice breaking change

We are changing the logic that creates `uid` in `LogRowModel`. Previously, for `uid` we used `id` field from log's data frame. Unfortunately, when users run multiple queries that returned duplicate logs data, `uid` was not unique which was causing bugs. To make `uid` unique across multiple queries that return duplicate logs data, we are now prepending `uid` with `refId` of query that produced the log line. We recommend not to rely on `LogRowModel` `uid` and instead use `dataFrame` `id` field.

### CODE REVIEW

File: public/app/core/logsModel.ts [hunk 1]
```diff
@@ -459,7 +459,8 @@ export function logSeriesToLogsModel(logSeries: DataFrame[], queries: DataQuery[
         entry,
         raw: message,
         labels: labels || {},
-        uid: idField ? idField.values[j] : j.toString(),
+        // Append refId to uid to make it unique across all series in a case when series contain duplicates
+        uid: `${series.refId}_${idField ? idField.values[j] : j.toString()}`,
```
Comment (svennergr) [inline public/app/core/logsModel.ts:463]: I am curious if we should append the `refId` in order to generate same IDs at least for Loki with that change?

I don't have a strong preference, just wanted to raise that.

Comment (ivanahuckova) [reply public/app/core/logsModel.ts:463]: You are right!

Comment (ivanahuckova) [reply public/app/core/logsModel.ts:463]: Fixed in https://github.com/grafana/grafana/pull/68569/commits/f2991c4b0a633c1d8446bbfd8e19fc3970a21ec6. I like `refId` being the first, so I updated lokiLiveStream code

**Review by gabor — APPROVED** [review]
LGTM 👍

File: public/app/core/logsModel.ts [hunk 2]
```diff
@@ -459,7 +459,8 @@ export function logSeriesToLogsModel(logSeries: DataFrame[], queries: DataQuery[
         entry,
         raw: message,
         labels: labels || {},
-        uid: idField ? idField.values[j] : j.toString(),
+        // Append refId to uid to make it unique across all series in a case when series contain duplicates
```
Comment (svennergr) [inline public/app/core/logsModel.ts:462]:
```suggestion
        // prepend refId to uid to make it unique across all series in a case when series contain duplicates
```

**Review by svennergr — APPROVED** [review]
LGTM

**Pull request merged**